tools/integration-test/src/tests/client_expiration.rs

```diff
@@ -0,0 +1,534 @@
+                "balance on wallet A should decrease",
+                &balance_a_2,
+                &(balance_a - 200),
+            )?;
+
+            let balance_b = chains.node_b.chain_driver().query_balance(
+                &chains.node_b.wallets().user1().address(),
+                &denom_b.as_ref(),
+            )?;
+
+            assert_eq("balance on wallet B should remain zero", &balance_b, &0)?;
+        }
+
+        Ok(())
+    }
+}
+
+impl BinaryChainTest for CreateOnExpiredClientTest {
+    fn run<ChainA: ChainHandle, ChainB: ChainHandle>(
+        &self,
+        _config: &TestConfig,
+        chains: ConnectedChains<ChainA, ChainB>,
+    ) -> Result<(), Error> {
+        // Create a connection before the IBC client expires, so that we can try create
+        // new channel with the connection after the client expired.
+        let connection = {
+            let _refresh_task_a = spawn_refresh_client(chains.client_b_to_a.clone())
+                .ok_or_else(|| eyre!("expect refresh task spawned"))?;
+
+            let _refresh_task_b = spawn_refresh_client(chains.client_a_to_b.clone())
+                .ok_or_else(|| eyre!("expect refresh task spawned"))?;
+
+            bootstrap_connection(&chains.client_b_to_a, &chains.client_a_to_b, false)?
+        };
+
+        wait_for_client_expiry();
+
+        info!("trying to bootstrap connection after IBC client is expired");
+
+        let res = bootstrap_connection(&chains.client_b_to_a, &chains.client_a_to_b, false);
+        match res {
+            Ok(_) => return Err(eyre!("expected bootstrap_connection to fail")),
+            Err(e) => {
+                info!("bootstrap_connection failed with expected error {}", e);
+            }
+        }
+
+        sleep(Duration::from_secs(5));
+
+        info!("trying to bootstrap channel after IBC client is expired");
+
+        let res = bootstrap_channel_with_connection(
+            &chains.handle_a,
+            &chains.handle_b,
+            connection,
+            &DualTagged::new(&PortId::transfer()),
+            &DualTagged::new(&PortId::transfer()),
+            false,
+        );
+
+        match res {
+            Ok(_) => return Err(eyre!("expected bootstrap_channel_with_connection to fail")),
+            Err(e) => {
+                info!(
+                    "bootstrap_channel_with_connection failed with expected error {}",
+                    e
+                );
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl BinaryChainTest for MisbehaviorExpirationTest {
+    fn run<ChainA: ChainHandle, ChainB: ChainHandle>(
+        &self,
+        _config: &TestConfig,
+        chains: ConnectedChains<ChainA, ChainB>,
+    ) -> Result<(), Error> {
+        /*
+           This test reproduce the error log when a misbehavior task is
+           first started. The error arise when `detect_misbehaviour_and_submit_evidence`
+           is called with `None`, and the initial headers are already expired.
```

Comment:
@adizere @romac I added this manual test to reproduce the error log from the misbehavior detection. The key is that when `detect_misbehaviour_and_submit_evidence` is called with `None`, and some update client events contain expired headers, then the error log is shown. Note however the returned result is still `ValidClient`, which is weird.